hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -105,53 +138,112 @@ func validateNetworkCIDRs(hc *hyperv1.HostedCluster) field.ErrorList {
 		return errs
 	}
 
-	return compareCIDREntries(cidrEntries)
+	return validateNoCIDROverlap(cidrEntries)
 }
 
 func validateSliceNetworkCIDRs(hc *hyperv1.HostedCluster) field.ErrorList {
 	var cidrEntries []cidrEntry
 
 	for _, cidr := range hc.Spec.Networking.MachineNetwork {
-		ce := cidrEntry{(net.IPNet)(cidr.CIDR), *field.NewPath("spec.networking.MachineNetwork")}
+		ce := cidrEntry{(net.IPNet)(cidr.CIDR), *field.NewPath("hostedcluster.spec.networking.machineNetwork")}
 		cidrEntries = append(cidrEntries, ce)
 	}
 	for _, cidr := range hc.Spec.Networking.ServiceNetwork {
-		ce := cidrEntry{(net.IPNet)(cidr.CIDR), *field.NewPath("spec.networking.ServiceNetwork")}
+		ce := cidrEntry{(net.IPNet)(cidr.CIDR), *field.NewPath("hostedcluster.spec.networking.serviceNetwork")}
 		cidrEntries = append(cidrEntries, ce)
 	}
 	for _, cidr := range hc.Spec.Networking.ClusterNetwork {
-		ce := cidrEntry{(net.IPNet)(cidr.CIDR), *field.NewPath("spec.networking.ClusterNetwork")}
+		ce := cidrEntry{(net.IPNet)(cidr.CIDR), *field.NewPath("hostedcluster.spec.networking.clusterNetwork")}
 		cidrEntries = append(cidrEntries, ce)
 	}
 
-	return compareCIDREntries(cidrEntries)
+	return validateNoCIDROverlap(cidrEntries)
 }
 
-func validateHostedClusterCreate(hc *hyperv1.HostedCluster) error {
-	errs := validateNetworkCIDRs(hc)
-	errs = append(errs, validateSliceNetworkCIDRs(hc)...)
+// This validates the old network settings do not collide with the cluster network settings.
+func validateServiceNetworkNoOverlap(c client.Client, ctx context.Context, hc *hyperv1.HostedCluster) field.ErrorList {
+	var errs field.ErrorList
+	var cidrEntries []cidrEntry
 
-	return errs.ToAggregate()
+	serviceCIDR := hc.Spec.Networking.ServiceCIDR
+	if serviceCIDR == "" {
+		return errs
+	}
+
+	_, serviceNet, err := net.ParseCIDR(serviceCIDR)
+	if err != nil {
+		errs = append(errs, field.Invalid(field.NewPath("hostedcluster.spec.networking.serviceCIDR"), serviceCIDR, err.Error()))
+		return errs
+	} else {
+		ce := cidrEntry{*serviceNet, *field.NewPath("hostedcluster.spec.networking.serviceCIDR")}
+		cidrEntries = append(cidrEntries, ce)
+	}
+
+	network := &configv1.Network{ObjectMeta: metav1.ObjectMeta{Name: "network"}}
```

Comment:
The only reason to define `ObjectMeta` here would be to use `ObjectKeyFromObject` further down in the `Get` call like e.G. here: https://github.com/openshift/hypershift/blob/ee373afb2a099dcf0741e11cc5770ed345cb74df/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L262

Lets either do that or not define `ObjectMeta` here